src/pcre2.h.generic

```diff
@@ -464,6 +464,17 @@ released, the numbers must not be changed. */
 #define PCRE2_CONFIG_COMPILED_WIDTHS        14
 #define PCRE2_CONFIG_TABLES_LENGTH          15
 
+/* Optimization directives for pcre2_set_optimize().
+For binary compatibility, only add to this list; do not renumber. */
+
+#define PCRE2_NO_OPTIMIZATION     0
+#define PCRE2_FULL_OPTIMIZATION   1
```

Comment:
PCRE2_OPT_LEVEL_0 .. PCRE2_OPT_LEVEL_3

Reply:
Philip mentioned before he didn't like the "optimization level" (at least as a parameter), so I think it is sensible not to use "OPT_LEVEL" either in the names. I would had used instead PCRE2_OPTIMIZATION_NONE and PCRE2_OPTIMIZATION_FULL though

Reply:
PCRE2_OPTIMIZATION_{NONE,FULL} sounds good.

The beauty of this proposal is that intermediate levels can easily be added later, if someone can think of one which makes sense.

For example (just imagining here), maybe someone might be working in a memory-constrained environment and might want only optimizations which can be done without increasing memory requirements. We can easily add another optimization directive which does that.

Perhaps someone might want only optimizations which don't increase compile time much. We can easily add another optimization directive which does that.